### PR TITLE
Merge [feature/khoatgd/add-new-brand-full] into [main]

### PR DESCRIPTION
**Pull request by trankhoa798**
PR description:

Add feature to create new brand:

1. Update dao
2. Update service
3. Update controller

14/11/2021

### CODE REVIEW

File: src/main/java/com/fa/training/dto/AdvertisingContractDto.java [hunk 1]
```diff
@@ -76,4 +76,15 @@ public AdvertisingContract dtoAsPersistent() {
 		return a;
 	}
 
+	public AdvertisingContractDto transformPersistentToDto(final AdvertisingContract contract) {
```
Comment (jason103nguyen) [inline src/main/java/com/fa/training/dto/AdvertisingContractDto.java:79]: I think you should be use the constructor for converting from entity to dto! What do you think about that?

File: src/main/java/com/fa/training/dto/BrandDto.java [hunk 2]
```diff
@@ -58,16 +58,29 @@ public void setBrandContracts(Set<AdvertisingContractDto> brandContracts) {
 	public Brand dtoAsPersistent() {
```
Comment (jason103nguyen) [inline src/main/java/com/fa/training/dto/BrandDto.java:58]: Please comment the purpose of this method!

**Review by jason103nguyen — APPROVED** [review]
Please check your code again and let's merge to main. Thanks!

File: src/main/java/com/fa/training/service/impl/BrandServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import com.fa.training.dao.BrandDaoImpl;
+import com.fa.training.dao.FootballPlayerDaoImpl;
+import com.fa.training.entities.Brand;
+import com.fa.training.service.common.AbstractService;
+import com.fa.training.service.common.IOperations;
+
+@Service
+public class BrandServiceImpl extends AbstractService<Brand> implements BrandService {
+
+	BrandDaoImpl brandDao;
```
Comment (MaHai1993) [inline src/main/java/com/fa/training/service/impl/BrandServiceImpl.java:20]: properties MUST be private.

File: src/main/java/com/fa/training/controller/BrandController.java [hunk 4]
```diff
@@ -1,4 +1,30 @@
...
+	public BrandController(BrandService brandService) {
+		super();
+		this.brandService = brandService;
+	}
+
+	@GetMapping(value = "/add-brand")
+	public String doGetAddBrand() {
+
+		BrandDto dto = new BrandDto();
+		dto.setName("Nike");
```
Comment (MaHai1993) [inline src/main/java/com/fa/training/controller/BrandController.java:26]: remove this hard code later

**Review by MaHai1993 — CHANGES_REQUESTED** [review]
Fix comment.
I help you change branch name in PR also.

**Pull request merged**